doc/arch/adr-003-collections-api.md

```diff
@@ -0,0 +1,82 @@
+
+The difference in `develop` and staging are because of two factors: staging workers are more powerful, and have much faster access to S3 tiles. These advantages should carry over to the Collections API variant as well, thus making it even faster when deployed.
+
+In addition, this will make our code simpler to understand and maintain, as well as make provisioning and deployment simpler by removing Spark and Spark JobServer.
+
+These improved speeds may also allow us to increase the area of interest threshold on the front-end, allowing users to select even bigger areas than before and get faster results back.
+
+## Challenges
+
+GeoTrellis 1.0 is compiled for Java 8, whereas the current MMW environment is setup with Java 7. We will have to make a number of updates to the MMW environment to support Java 8.
+
+Since Spark JobServer is no longer used, we will have to switch to using Akka HTTP for exposing the Geoprocessing service. This will also require updates to the relevant geoprocessing tasks in Celery.
+
+Changes to the geoprocessing service may result in differences in how cells are counted, and may lead to different results for the same inputs. This is already seen to some extent in [mmw-geoprocessing#44](https://github.com/WikiWatershed/mmw-geoprocessing/pull/44). It is difficult to say which is _more_ correct, the current or the new implementation. But we will have to explain and justify the changes.
+
+The Collections API has so far never been deployed to production. It comes with risks of "unknown unknowns", and we may have to solve unforeseen technical challenges.
+
+## Drawbacks
+
+Removing Spark JobServer will lead to the interaction between Celery and Geoprocessing becoming synchronous rather than asynchronous. This should be a tolerable difference, since the new synchronous requests should be 3–5 times faster than the old asynchronous ones, and a Celery task waiting for 5-10 seconds should be manageable.
+
+The Collections API supports far fewer MapAlgebra operations than the RDD API. This is okay for now, as we are only counting cells and averaging their values. However, if we were to require MapAlgebra operations in the future, we may have to add back the RDD API.
+
+## Approach
+
+ 1. Make a set of use cases that cover all the functionality of the current geoprocessing service
+ 2. Make a new project within the geoprocessing repo, and add an Akka HTTP service
+ 3. Add endpoints that take in the same JSON input as the current service, and perform the same operations, just using the Collections API instead
+ 4. Fine tune results so they match with the previous ones, or explain and justify why they are different
+ 5. Deploy the new geoprocessing service to Quay
+ 6. Update provisioning to use the new service
+ 7. Update geoprocessing Celery tasks to use the new service
+ 8. Test the UI to ensure values are still similar and sensible
+ 9. Remove Spark and Spark JobServer from provisioning
```

Comment:
I'm sure if we go down this route, we'll have to spend some time cleaning, updating and changing the related dependencies.  I think it's enough to know that it would be a sufficiently large change to our provisioning, and we would not be operating a spark local or cluster after the fact, whether or not it remained an application dependency.